c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1150,14 +1150,14 @@ static std::string reportProcessMemoryInfo(int device) {
           pci_id, &nvml_device));
 
   std::vector<nvmlProcessInfo_v1_t> procs(8);
-  unsigned int size = procs.size();
+  auto size = procs.size();
   nvmlReturn_t r;
   while ((r = DriverAPI::get()->nvmlDeviceGetComputeRunningProcesses_(
               nvml_device, &size, procs.data())) ==
          NVML_ERROR_INSUFFICIENT_SIZE) {
     procs.resize(size);
   }
-  unsigned int self_pid = getpid();
+  pid_t self_pid = getpid();
```

Comment:
This is causing a build error because ti's being compared with unsigned int later in the method.

Reply:
Weird, `getpid` is `pid_t` everywhere:
- Linux: https://man7.org/linux/man-pages/man2/getpid.2.html https://linux.die.net/man/2/getpid
- POSIX: https://pubs.opengroup.org/onlinepubs/009696699/functions/getpid.html
- zOS: https://www.ibm.com/docs/en/zos/2.4.0?topic=functions-getpid-get-process-id
- FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=getpid&sektion=2